src/skUnit/Scenarios/Parsers/Assertions/FunctionCallCheckAssertion.cs

```diff
@@ -103,24 +102,23 @@ public async Task Assert(Semantic semantic, string input, IEnumerable<object>? h
 
                 if (FunctionArguments.Any())
                 {
-                    var lastFunctionCall = thisFunctionCalls.Last();
+                    var thisFunctionCall = thisFunctionCalls.Last();
 
-                    var relatedCall = (
-                        from call in chatMessageHistory.SelectMany(c => c.Contents).OfType<FunctionCallContent>()
-                        where call.CallId == lastFunctionCall.CallId
-                        select call
+                    var thisCallResult = (
+                        from fr in chatMessageHistory.SelectMany(c => c.Contents).OfType<FunctionResultContent>()
+                        where fr.CallId == thisFunctionCall.CallId
+                        select fr
                     ).FirstOrDefault();
 
-                    if (relatedCall is null)
+                    if (thisCallResult is null)
                         throw new SemanticAssertException(
                             $"""
-                             No function call found with name: {FunctionName}
-                             Current calls: {string.Join(", ", functionCalls.Select(fc => fc.Name))}
```

Comment:
I saw your tweet:
> FunctionResultContent no longer has a Name property.
> ([@stephentoub](https://x.com/stephentoub)
, 
[@stevensanderson](https://x.com/stevensanderson)
 Why!? Was I the only one using it! 😅)

Is this the only place you were using it?

It was removed because it was duplicative information (it's in the corresponding FunctionCallContent with the same tool call ID) that no known IChatClient implementations actually used. The relevant piece of information is the tool call ID, and if you really want the function name and/or arguments associated with that ID, you can look for the FunctionCallContent that maps to the ID.